### PR TITLE
Make zero in place

### PR DESCRIPTION
**Pull request by wsmoses**
It's 2 am so I'lls tart this later, but something like this.

cc @ChrisRackauckas

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 [compat]
 CEnum = "0.4, 0.5"
 ChainRulesCore = "1"
-EnzymeCore = "0.7.3"
+EnzymeCore = "0.7.3, 0.7.4"
```
Comment (vchuravy) [inline Project.toml:22]: This needs to be `0.7.4` since otherwise you will get a "EnzymeCore.make_zero!" not defined.

Comment (wsmoses) [reply Project.toml]: i was just wondering this

**Comment by ChrisRackauckas** [comment]
I feel like make_zero and make_zero! should be moved out of Enzyme as a general Julia tool, even to Base as the standard `zero` function extension. I kind of just use it everywhere...

**Comment by wsmoses** [comment]
Not opposed, but have to start somewhere.

**Pull request merged**